daemon/logger/fluentd/fluentd_test.go

```diff
@@ -199,3 +205,118 @@ func TestValidateLogOptAddress(t *testing.T) {
 		}
 	}
 }
+
+func TestValidateWriteTimeoutDuration(t *testing.T) {
+	invalidDurations := []string{"-1", "1", "-1s"}
+	for _, d := range invalidDurations {
+		t.Run("invalid "+d, func(t *testing.T) {
+			err := ValidateLogOpt(map[string]string{writeTimeoutKey: d})
+			assert.ErrorContains(t, err, "invalid value for fluentd-write-timeout:")
+		})
+	}
+
+	validDurations := map[string]time.Duration{
+		"100ms": 100 * time.Millisecond,
+		"10s":   10 * time.Second,
+		"":      0,
+	}
+	for k, v := range validDurations {
+		t.Run("valid "+k, func(t *testing.T) {
+			err := ValidateLogOpt(map[string]string{writeTimeoutKey: k})
+			assert.NilError(t, err)
+			// Ignore err here since that's already checked in the ValidateLogOpt error check.
+			cfg, _ := parseConfig(map[string]string{writeTimeoutKey: k})
+			assert.Equal(t, cfg.WriteTimeout, v)
+		})
+	}
+}
+
+// TestWriteTimeoutIsEffective tests that writes timeout when the server is unresponsive.
+// The test ensures that instead of hanging forever, the fluentd write operation returns
+// an error when writes cannot be completed within the specified duration.
+func TestWriteTimeoutIsEffective(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("windows not supported")
+	}
+
+	socketFile := "/tmp/fluent-logger-golang.sock"
+	l, err := net.Listen("unix", socketFile)
+	assert.NilError(t, err, "unable to create listener for socket %s", socketFile)
+	defer l.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
I was trying the test with the option disabled, and that caused it (expectedly) to hang forever; perhaps we should add a timeout here .. just in case?


```go
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Reply:
sure, sounds good.